examples/connext_dds/dynamic_data_request_reply/cpp/PrimeNumberReplier.cxx

```diff
@@ -0,0 +1,328 @@
+         */
+
+        const DDS::Duration_t MAX_WAIT = {20, 0};
+
+        /* receive_request returns false when it times out */
+        connext::LoanedSamples<DDS_DynamicData>received_requests = 
+                replier.receive_requests(MAX_WAIT);
+
+        bool stop=1;
+
+        while(stop) {
+
+            typedef connext::LoanedSamples<DDS_DynamicData>::iterator iterator;
+            for (
+                    iterator request_sample = received_requests.begin(); 
+                    request_sample != received_requests.end(); 
+                    ++request_sample) 
+            {
+                
+                if(!request_sample->info().valid_data) {
+                    continue;
+                }
+                
+                /* We obtain "n" and "primes_per_reply" 
+                 * from the received sample:
+                 */
+                
+                DDS_Long n, primes_per_reply; 
+
+                request_sample->data().get_long(
+                        n,
+                        "n",
+                        DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED);
+
+                request_sample->data().get_long(
+                        primes_per_reply,
+                        "primes_per_reply",
+                        DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED);
+
+                /* This constant is defined in PrimesType.cxx */
+                if (n <= 0 || primes_per_reply <= 0 
+                        || primes_per_reply > THE_PRIME_SEQUENCE_MAX_LENGTH)
+                {
+
+                    std::cout << "Cannot process request" << std::endl;
+
+                    /* Send reply indicating error */
+                    reply->set_long(
+                            "status",
+                            DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,
+                            2);
+
+                    replier.send_reply(
+                            *reply, 
+                            request_sample->identity());
+                }
+                else{
+                    std::cout << "Calculating prime numbers below "
+                          << n << "... " << std::endl;
+                }
+
+                calculate_and_send_primes(
+                        replier,
+                        request_sample->identity(),
+                        n,
+                        primes_per_reply,
+                        reply);
+
+                std::cout << "DONE" << std::endl;
+ 
+            }
+
+            stop=0;
+        }
+    }
+};
+
+int replier_main(int domain_id) {
+    /* Uncomment this to turn on additional logging
+        NDDSConfigLogger::get_instance()->set_verbosity(
+            NDDS_CONFIG_LOG_VERBOSITY_WARNING);
+    */
+    std::cout << "PrimeNumberReplier running (on domain " << domain_id << ")"
+              << std::endl;
+
+    try {
+        /*
+         * Run the example
+         */
+        PrimeNumberReplierExample(domain_id).run_example();
```

Comment:
this function (replier_main) should contain the example instead of run_example()